src/main/io/gps.c

```diff
@@ -573,110 +846,259 @@ static void ubloxSetMessageRate(uint8_t messageClass, uint8_t messageID, uint8_t
+        payload[0] = gpsConfig()->sbasMode != SBAS_NONE;
+
+        size_t offset = ubloxValSet(&tx_buffer, CFG_SBAS_USE_TESTMODE, payload, UBX_VAL_LAYER_RAM);
+
+        payload[0] = 1;
+        offset += ubloxAddValSet(&tx_buffer, CFG_SBAS_USE_RANGING, payload, offset);
+        offset += ubloxAddValSet(&tx_buffer, CFG_SBAS_USE_DIFFCORR, payload, offset);
 
-    //NOTE: default ublox config for sbas mode is: UBLOX_USAGE_RANGE | UBLOX_USAGE_DIFFCORR, integrity is disabled
-    tx_buffer.payload.cfg_sbas.usage = UBLOX_USAGE_RANGE | UBLOX_USAGE_DIFFCORR;
-    if (gpsConfig()->sbas_integrity) {
-        tx_buffer.payload.cfg_sbas.usage |= UBLOX_USAGE_INTEGRITY;
+        if (gpsConfig()->sbas_integrity) {
+            offset += ubloxAddValSet(&tx_buffer, CFG_SBAS_USE_INTEGRITY, payload, offset);
+        }
+
+        uint64_t mask = SBAS_SEARCH_ALL;
+        switch (gpsConfig()->sbasMode) {
+            case SBAS_EGNOS:
+                mask = SBAS_SEARCH_PRN(123) | SBAS_SEARCH_PRN(126) | SBAS_SEARCH_PRN(136);
+                break;
+            case SBAS_WAAS:
+                mask = SBAS_SEARCH_PRN(131) | SBAS_SEARCH_PRN(133) | SBAS_SEARCH_PRN(135) | SBAS_SEARCH_PRN(138);
+                break;
+            case SBAS_MSAS:
+                mask = SBAS_SEARCH_PRN(129) | SBAS_SEARCH_PRN(137);
+                break;
+            case SBAS_GAGAN:
+                mask = SBAS_SEARCH_PRN(127) | SBAS_SEARCH_PRN(128) | SBAS_SEARCH_PRN(132);
+                break;
+            case SBAS_AUTO:
+            default:
+                break;
+        }
+
+        payload[0] = (uint8_t)(mask >> (8 * 0));
+        payload[1] = (uint8_t)(mask >> (8 * 1));
+        payload[2] = (uint8_t)(mask >> (8 * 2));
+        payload[3] = (uint8_t)(mask >> (8 * 3));
+        payload[4] = (uint8_t)(mask >> (8 * 4));
+        payload[5] = (uint8_t)(mask >> (8 * 5));
+        payload[6] = (uint8_t)(mask >> (8 * 6));
+        payload[7] = (uint8_t)(mask >> (8 * 7));
+
+        offset += ubloxAddValSet(&tx_buffer, CFG_SBAS_PRNSCANMASK, payload, offset);
+
+        ubloxSendConfigMessage(&tx_buffer, MSG_CFG_VALSET, offsetof(ubxCfgValSet_t, cfgData) + offset, true);
+    } else {
+        //NOTE: default ublox config for sbas mode is: UBLOX_MODE_ENABLED, test is disabled
+        tx_buffer.payload.cfg_sbas.mode = UBLOX_MODE_TEST;
+        if (gpsConfig()->sbasMode != SBAS_NONE) {
+            tx_buffer.payload.cfg_sbas.mode |= UBLOX_MODE_ENABLED;
+        }
+
+        // NOTE: default ublox config for sbas mode is: UBLOX_USAGE_RANGE | UBLOX_USAGE_DIFFCORR, integrity is disabled
+        tx_buffer.payload.cfg_sbas.usage = UBLOX_USAGE_RANGE | UBLOX_USAGE_DIFFCORR;
+        if (gpsConfig()->sbas_integrity) {
+            tx_buffer.payload.cfg_sbas.usage |= UBLOX_USAGE_INTEGRITY;
+        }
+
+        tx_buffer.payload.cfg_sbas.maxSBAS = 3;
+        tx_buffer.payload.cfg_sbas.scanmode2 = 0;
+        switch (gpsConfig()->sbasMode) {
+            case SBAS_AUTO:
```

Comment:
(table for valset can be probably reused here)

Reply:
need a simple code suggestion to help me achieve that :-)